src/App.tsx

```diff
@@ -1,61 +1,118 @@
 import './App.scss';
+import React, { useState } from 'react';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+const MY_REGEX = /[^a-z-а-я-0-9-' ']/gi;
 
 export const App = () => {
+  const [todos, setTodos] = useState(todosFromServer);
+  const [userId, setUserId] = useState(0);
+  const [title, setTitle] = useState('');
+  const [errorTitle, setErrorTitle] = useState(false);
+  const [errorUser, setErrorUser] = useState(false);
+  const [counter, setCounter] = useState(
+    Math.max(...todos.map(({ id }) => id)) + 1,
+  );
+
+  const isValidData = () => {
+    if (!userId) {
+      setErrorUser(true);
+    }
+
+    if (!title.trim()) {
+      setErrorTitle(true);
+    }
+
+    return !!userId && !!title;
+  };
+
+  const resetFields = () => {
+    setUserId(0);
+    setTitle('');
+  };
+
+  const addUser = () => {
+    setTodos((prevState) => (
+      [
+        ...prevState,
+        {
+          id: counter,
+          title,
+          completed: false,
+          userId,
+        }]));
+  };
+
+  const handleSubmitButton = (event: React.MouseEvent<HTMLButtonElement>) => {
+    event.preventDefault();
+    if (!isValidData()) {
+      return;
+    }
+
+    addUser();
+    resetFields();
+    setCounter(state => state + 1);
+  };
+
+  const handleTitleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(event.target.value.replace(MY_REGEX, ''));
```

Comment:
Why you need this replace() ?

Reply:
It`s an optional task:  "Allow entering only letters (ru and en), digits, and spaces in the title."